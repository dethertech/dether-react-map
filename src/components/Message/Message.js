import React from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import tokens from '../../styles/tokens'

const Wrapper = styled.div`
  display: flex;
  margin: ${tokens.spaces.xs} 0;
  justify-content: center;

  ${({ alignLeft }) => 
    alignLeft &&
    css`
      text-align: left;
    `};

  ${({ theme }) => {
    switch (theme) {
      case 'infored':
        return css`
          color: ${tokens.colors.red};
          font-size: ${tokens.fontSizes.xxxs};
          font-style: italic;
          font-weight: 500;
        `
      case 'info':
        return css`
          color: ${tokens.colors.grey.light};
          font-size: ${tokens.fontSizes.xxxs};
          font-style: italic;
          font-weight: 300;
        `
      case 'danger':
        return css`
          color: ${tokens.colors.red};
          font-size: ${tokens.fontSizes.xxxs};
          font-weight: 500;
        `
      default:
        return css`
          color: ${tokens.colors.black};
          font-size: ${tokens.fontSizes.xs};
        `
    }
  }};
`

const Icon = styled.div`
  font-size: ${tokens.fontSizes.xxs};
  color: ${tokens.colors.white};
  width: 1.5rem;
  height: 1.5rem;
  flex: 0 0 1.5rem;
  overflow: hidden;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: space - around;
  font-weight: 900;
  border-radius: ${tokens.radius.circle};
  margin-right: ${tokens.spaces.s};

  ${({ theme }) => {
    switch (theme) {
      case 'info':
        return css`
          background-color: ${tokens.colors.blue};
          &:before {
            content: 'i';
          }
        `
      case 'danger':
        return css`
          background-color: ${tokens.colors.red};
          &:before {
            content: '!';
          }
        `
      default:
        return css`
          background-color: ${tokens.colors.black};
          &:before {
            content: '!';
          }
        `
    }
  }};
`

const Message = ({ theme, withIcon, alignLeft, children, onClick }) => (
  <Wrapper
    theme={theme}
    withIcon={withIcon}
    alignLeft={alignLeft}
    onClick={onClick}
  >
    {withIcon && <Icon theme={theme} />}
    <div>{children}</div>
  </Wrapper>
)

Message.propTypes = {
  theme: PropTypes.oneOf(['info', 'error', 'danger']),
  withIcon: PropTypes.bool,
  alignLeft: PropTypes.bool,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
}

Message.defaultProps = {
  alignLeft: false,
  theme: 'info',
  withIcon: true,
  onClick: undefined,
}

export default Message
