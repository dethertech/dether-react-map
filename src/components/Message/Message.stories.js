/* eslint-disable */
import React from 'react'
import styled, { css } from 'styled-components'
import { storiesOf } from '@storybook/react'

import Message from './index'

storiesOf('03 - Components', module)
.addWithChapters('Messages', {
  subtitle: 'Message error',
  chapters: [
    {
      sections: [
      {
        title: 'Select Input',
        sectionFn: () => (
          <Message theme="error">
            Hello je suis une erreur. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Consequuntur quaerat, doloremque neque voluptas ut laudantium similique nulla consequatur
            nesciunt labore enim, at numquam dolor fugit maxime commodi facere odio cupiditate.
          </Message>
        ),
        options: {
          showSource: true,
          showPropTables: true
        }
      },{
        title: 'Select Input',
        sectionFn: () => (
          <Message theme="info">
            Hello je suis un Message. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Consequuntur quaerat, doloremque neque voluptas ut laudantium similique nulla consequatur
            nesciunt labore enim, at numquam dolor fugit maxime commodi facere odio cupiditate.
          </Message>
        ),
        options: {
          showSource: true,
          showPropTables: true
        }
      },{
        title: 'Select Input',
        sectionFn: () => (
          <Message withIcon={false} >
            Hello je suis un Message sans icone. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
            Consequuntur quaerat, doloremque neque voluptas ut laudantium similique nulla consequatur
            nesciunt labore enim, at numquam dolor fugit maxime commodi facere odio cupiditate.
          </Message>
        ),
        options: {
          showSource: true,
          showPropTables: true
        }
      }
    ]
  }]
});
