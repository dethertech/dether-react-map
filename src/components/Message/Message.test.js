/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';


import Message from './index';



describe('Component Message', () => {
  const component = renderer
    .create(
      <Message theme="info">
        Hello je suis un Message. Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        Consequuntur quaerat, doloremque neque voluptas ut laudantium similique nulla consequatur
        nesciunt labore enim, at numquam dolor fugit maxime commodi facere odio cupiditate.
      </Message>
    );
  const tree = component.toJSON();

  it('should be render without crash', () => {
    expect(tree).toMatchSnapshot();
  });
});
