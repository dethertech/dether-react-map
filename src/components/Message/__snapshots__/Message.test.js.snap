// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Message should be render without crash 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1.6rem 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #121E38;
  font-size: 1.2rem;
}

.c1 {
  font-size: 1rem;
  color: #FFFFFF;
  width: 1.5rem;
  height: 1.5rem;
  -webkit-flex: 0 0 1.5rem;
  -ms-flex: 0 0 1.5rem;
  flex: 0 0 1.5rem;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space - around;
  -webkit-justify-content: space - around;
  -ms-flex-pack: space - around;
  justify-content: space - around;
  font-weight: 900;
  border-radius: 50%;
  margin-right: 1.2rem;
  background-color: #3C80F1;
}

.c1:before {
  content: 'i';
}

<div
  className="c0"
  onClick={undefined}
>
  <div
    className="c1"
  />
  <div>
    Hello je suis un Message. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequuntur quaerat, doloremque neque voluptas ut laudantium similique nulla consequatur nesciunt labore enim, at numquam dolor fugit maxime commodi facere odio cupiditate.
  </div>
</div>
`;
